packages/runtime/garbage-collector/src/utils.ts

```diff
@@ -69,10 +69,13 @@ export function cloneGCData(gcData: IGarbageCollectionData): IGarbageCollectionD
         return childGCDetailsMap;
     }
 
-    // Remove the node's self GC nodes, if any, and generate the children GC nodes.
     const gcNodes = gcDetails.gcData.gcNodes;
-    delete gcNodes["/"];
     for (const [id, outboundRoutes] of Object.entries(gcNodes)) {
+        // Skip self-node since only children GC data is to be generated.
+        if (id === "/") {
+            continue;
+        }
```

Comment:
> Fixed a bug in `unpackChildNodesGCDetails` API which was changing the passed GC data be removing node with id "/".

Are you saying that you're not passing down the parent GC data to the child as it causes a bug of sorts? Is the bug that the child has the wrong GC data?

Reply:
No, what I am saying is that this API was changing the gcDetails that was passed to it by deleting a node. Imagine that a data store called this API on its GC details to get its children GC details. This API will remove the "/" node and the GC details in the data store is now incorrect.
It worked until now because the places where it was called never used the GC details it passed to this API again. But with this change, the GC details are used later.

Reply:
I see